sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -388,99 +397,98 @@ object SQLConf {
     .internal()
     .doc("Adaptive query execution is skipped when the query does not have exchanges or " +
       "sub-queries. By setting this config to true (together with " +
-      s"'${ADAPTIVE_EXECUTION_ENABLED.key}' enabled), Spark will force apply adaptive query " +
+      s"'${ADAPTIVE_EXECUTION_ENABLED.key}' set to true), Spark will force apply adaptive query " +
       "execution for all supported queries.")
     .version("3.0.0")
     .booleanConf
     .createWithDefault(false)
 
-  val REDUCE_POST_SHUFFLE_PARTITIONS_ENABLED =
-    buildConf("spark.sql.adaptive.shuffle.reducePostShufflePartitions")
-      .doc(s"When true and '${ADAPTIVE_EXECUTION_ENABLED.key}' is enabled, this enables reducing " +
-        "the number of post-shuffle partitions based on map output statistics.")
+  val ADVISORY_SHUFFLE_PARTITION_SIZE_IN_BYTES =
+    buildConf("spark.sql.adaptive.advisoryShufflePartitionSizeInBytes")
+      .doc("The advisory size in bytes of the shuffle partition during adaptive optimization. " +
+        "It takes effect when Spark coalesces small shuffle partitions or splits skewed shuffle " +
+        "partition.")
       .version("3.0.0")
-      .booleanConf
-      .createWithDefault(true)
+      .fallbackConf(SHUFFLE_TARGET_POSTSHUFFLE_INPUT_SIZE)
 
-  val FETCH_SHUFFLE_BLOCKS_IN_BATCH_ENABLED =
-    buildConf("spark.sql.adaptive.shuffle.fetchShuffleBlocksInBatch")
-      .doc("Whether to fetch the continuous shuffle blocks in batch. Instead of fetching blocks " +
-        "one by one, fetching continuous shuffle blocks for the same map task in batch can " +
-        "reduce IO and improve performance. Note, multiple continuous blocks exist in single " +
-        s"fetch request only happen when '${ADAPTIVE_EXECUTION_ENABLED.key}' and " +
-        s"'${REDUCE_POST_SHUFFLE_PARTITIONS_ENABLED.key}' is enabled, this feature also depends " +
-        "on a relocatable serializer, the concatenation support codec in use and the new version " +
-        "shuffle fetch protocol.")
+  val COALESCE_SHUFFLE_PARTITIONS_ENABLED =
+    buildConf("spark.sql.adaptive.coalesceShufflePartitions.enabled")
```

Comment:
Can we make these shorter and easier to remember? Personally,  maybe change `spark.sql.adaptive.coalesceShufflePartitions.enabled` to `spark.sql.adaptive.mergePartitions`, `spark.sql.adaptive.coalesceShufflePartitions.initialPartitionNum` to `spark.sql.adaptive.mergePartitions.initialNum` and `spark.sql.adaptive.coalesceShufflePartitions.minPartitionNum` to `spark.sql.adaptive.mergePartiontions.minNum`.

Reply:
Shuffle is where this behavior happens and have been written in the doc field, we may not need to enforce it in the config name, and it seems that we do not have any other places under `adaptive` to coalesce partitions. And merge might be easier to spell than coalesce :)

Reply:
I'm fine with "coalescePartitions".

Reply:
shall we also rename `advisoryShufflePartitionSizeInBytes` to `advisoryPartitionSizeInBytes`?

Reply:
`advisoryPartitionSizeInBytes` looks good to me.

Reply:
Can we uniform the verb naming in `reduce`, `coalesce`, or `merge `both in here configuration name and the optimization rule name  of `ReduceNumShufflePartitions` ? If we use the `coalescePartitions`, It is better to modify the optimization rule name from `ReduceNumShufflePartitions`  to `CoalesceShufflePartitions` ?

Reply:
Think we should, but we can do it in another PR. The code name is not user facing and doesn't need to made into 3.0.